[BME-5] Added a check make sure setS3Auth is run prior to running Zep start

dustinvanstee opened the pull request:
Added a status flag to make sure setS3Auth is run before z.start.  If not, an error message will print.  Also added a simple check for making sure rootkey.csv exists.


added a new status flag s3authSet = False
once setS3Auth is run, then this flag is set True
start method checks this flag prior to start.   if not set, prints
error message and aborts ..

dustinvanstee committed:
Added a check make sure setS3Auth is run prior to running Zep start

added a new status flag s3authSet = False
once setS3Auth is run, then this flag is set True
start method checks this flag prior to start.   if not set, prints
error message and aborts ..

rawkintrevo commented on data/services/zeppelin.py:
```diff
@@ -34,19 +35,25 @@ class ZeppelinServiceOnBI(AbstractServiceOnBI):
+                    print stderr.read()
+        else :
+            print " ERROR : need to run setS3auth method prior to running start method"
+            exit()
```
+1 on this, much more elegent

dustinvanstee replied:
performed some simple manual tests ...
1.  tested is s3setAuth not run prior to zeppelin start
2.  tested with rootkey.csv file missing 
both tests passed as expected

rawkintrevo commented on data/services/zeppelin.py:
```diff
@@ -140,26 +147,35 @@ def addMahoutConfig(self, terpName = None, configs={}):
+    
+            self.s3authSet = True
+        else : 
+            print "ERROR : rootkey.csv file not detected.  Go to AWS security center and generate access key.  Save in data/resources/aws/rootkey.csv"
```
I think before it would have failed on a 'IOException File Not Found' for rootkey.csv not being present (though it might not have deadlined the program.  At anyrate- we should raise an exception here, not simply (and very quietly) print an error message on what will eventually be a catastrophic issue.

dustinvanstee replied:
+1  I will add that and re-issue pr

rawkintrevo replied:
if you just recommit the pr will update automatically

dustinvanstee committed:
Fixed file exists check for rootkey.cvs

added a try/except statement and exit gracefully.

dustinvanstee committed:
Merge remote-tracking branch 'rawkintrevo/master' into BME-5

rawkintrevo commented on data/services/zeppelin.py:
```diff
@@ -34,19 +35,25 @@ class ZeppelinServiceOnBI(AbstractServiceOnBI):
-        if "FAILED" in zeppelin_status:
-            stdin, stdout, stderr = self.ssh.exec_command(self.dirName + "/bin/zeppelin-daemon.sh start")
+        # Add quick check here to see if s3AuthSet was run
+        if(self.s3authSet) : 
```
This introduces API breaking changes.  The example scripts need to be updated from:

```python
z = ZeppelinServiceOnBI(SERVER, USERNAME, PASSWORD)
z.install()
z.start()
sleep(5)
z.downloadConfig({"interpreter.json"  : "conf/interpreter.json"})
z.updateConfig()
#z.setS3auth(S3_USERNAME, S3_BUCKET) ## Must do this first!
```

to 

```python
z = ZeppelinServiceOnBI(SERVER, USERNAME, PASSWORD)
z.install()
z.setS3auth(S3_USERNAME, S3_BUCKET) ## Must do this first!
z.start()
sleep(5)
z.downloadConfig({"interpreter.json"  : "conf/interpreter.json"})
z.updateConfig()
```

E.g. moving `setS3auth(...)` to before the first `z.start()`

rawkintrevo commented on data/services/zeppelin.py:
```diff
@@ -140,7 +147,13 @@ def addMahoutConfig(self, terpName = None, configs={}):
+        try :
+            aws_keys = open(root_key_path).readlines()
+        except (IOError):
+            print "ERROR : rootkey.csv file not detected.  Go to AWS security center and generate access key.  Save in data/resources/aws/rootkey.csv"
```
add `or specify alternate path`

rawkintrevo commented on data/services/zeppelin.py:
```diff
@@ -140,7 +147,13 @@ def addMahoutConfig(self, terpName = None, configs={}):
+            aws_keys = open(root_key_path).readlines()
+        except (IOError):
+            print "ERROR : rootkey.csv file not detected.  Go to AWS security center and generate access key.  Save in data/resources/aws/rootkey.csv"
+            exit()
```
`exit(1)` to indicate failure.

rawkintrevo commented:
Looking good @dustinvanstee . A couple of more comments to address after testing then should be good to merge. 

Thanks!

dustinvanstee committed:
Addressed rawkintrevo's comments for PR BME-5

Re-ordered example scripts so they aren't broken, and added some minor
mod's to zeppelin.py

dustinvanstee commented:
Addressed comments above, and performed some tests using the example deployment scripts ... Looks good to me.  Code re-committed

rawkintrevo commented on example_zeppelin_mahout.py:
```diff
@@ -48,13 +49,8 @@
 z.start()
 z.setupR()
 
-<<<<<<< HEAD
```
Need to clean up the merge

rawkintrevo commented:
lgtm if no further comment
